projects/lights_control/src/lights_can.c

```diff
@@ -0,0 +1,45 @@
+#include "can.h"
+#include "can_hw.h"
+#include "can_msg_defs.h"
+#include "can_unpack.h"
+#include "event_queue.h"
+#include "status.h"
+
+#include "lights_can.h"
+#include "lights_events.h"
+
+// RX handler function.
+static StatusCode prv_rx_handler(const CANMessage *msg, void *context, CANAckStatus *ack_reply) {
+  const LightsCanSettings *settings = (const LightsCanSettings *)context;
+  // Unpacks the message, and raises events.
+  LightsCanActionID action_id = 0;
+  uint8_t data = 0;
+  CAN_UNPACK_LIGHTS_STATES(msg, (uint8_t *)&action_id, &data);
+  if (action_id >= NUM_LIGHTS_CAN_ACTIONS_ID) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Unsupported lights action.");
+  }
+  status_ok_or_return(
+      event_raise(settings->event_lookup[action_id][data], settings->event_data_lookup[action_id]));
```

Comment:
nit. weird line break

Reply:
@ckitagawa Is it safe to assume that `data` will be between `0` and `1`? I'm assuming that `0` means "off" and `1` means "on".